support-workers/src/main/scala/com/gu/acquisitions/AcquisitionDataRowBuilder.scala

```diff
@@ -0,0 +1,215 @@
+      case _: DirectDebitPaymentMethod | _: ClonedDirectDebitPaymentMethod => DirectDebit
+    }
+
+  private def getAbTests(data: AcquisitionData) =
+    (data.supportAbTests ++ data.referrerAcquisitionData.abTests.getOrElse(Set()))
+      .map(abTest => acquisitions.AbTest(
+        abTest.name,
+        abTest.variant
+      )).toList
+
+  private def productTypeAndAmount(product: ProductType): (AcquisitionProduct, Option[BigDecimal]) = product match {
+    case c: Contribution => (AcquisitionProduct.RecurringContribution, Some(c.amount.toDouble))
+    case _: DigitalPack => (AcquisitionProduct.DigitalSubscription, None)
+    case _: Paper => (AcquisitionProduct.Paper, None)
+    case _: GuardianWeekly => (AcquisitionProduct.GuardianWeekly, None)
+  }
+
+  private def printOptionsFromProduct(product: ProductType, deliveryCountry: Option[Country]): Option[PrintOptions] = {
+
+    def printProduct(fulfilmentOptions: FulfilmentOptions, productOptions: ProductOptions): PrintProduct =
+      (fulfilmentOptions, productOptions) match {
+        case (HomeDelivery, Everyday) => HomeDeliveryEveryday
+        case (HomeDelivery, EverydayPlus) => HomeDeliveryEverydayPlus
+        case (HomeDelivery, Sixday) => HomeDeliverySixday
+        case (HomeDelivery, SixdayPlus) => HomeDeliverySixdayPlus
+        case (HomeDelivery, Weekend) => HomeDeliveryWeekend
+        case (HomeDelivery, WeekendPlus) => HomeDeliveryWeekendPlus
+        case (HomeDelivery, Saturday) => HomeDeliverySaturday
+        case (HomeDelivery, SaturdayPlus) => HomeDeliverySaturdayPlus
+        case (HomeDelivery, Sunday) => HomeDeliverySunday
+        case (HomeDelivery, SundayPlus) => HomeDeliverySundayPlus
+        case (Collection, Everyday) => VoucherEveryday
+        case (Collection, EverydayPlus) => VoucherEverydayPlus
+        case (Collection, Sixday) => VoucherSixday
+        case (Collection, SixdayPlus) => VoucherSixdayPlus
+        case (Collection, Weekend) => VoucherWeekend
+        case (Collection, WeekendPlus) => VoucherWeekendPlus
+        case (Collection, Saturday) => VoucherSaturday
+        case (Collection, SaturdayPlus) => VoucherSaturdayPlus
+        case (Collection, Sunday) => VoucherSunday
+        case _ => VoucherSundayPlus
+      }
+
+    product match {
+      case p: Paper => Some(PrintOptions(
+       printProduct(p.fulfilmentOptions, p.productOptions),
+        Country.UK
+      ))
+      case _: GuardianWeekly => Some(PrintOptions(
+        PrintProduct.GuardianWeekly,
+        deliveryCountry.getOrElse(Country.UK)
+      ))
+      case _ => None
+    }
+  }
+
+  private def getAcquisitionTypeDetails(s: SendThankYouEmailState): AcquisitionTypeDetails =
+    s match {
+      case s: SendThankYouEmailContributionState => AcquisitionTypeDetails(
+        Some(s.paymentMethod),
+        None,
+        Direct,
+        Purchase,
+        Some(s.accountNumber),
+        None //TODO: we should have a sub number for this
```

Comment:
interesting - I stopped passing them through for products where they were not used, was that a regression?

Reply:
Well we didn't use them before but the Data Design team are keen to have them so they can join to the zuora export tables in the data lake